newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js

```diff
@@ -163,6 +163,15 @@ const PrivateAssetPackPurchaseDialog = ({
     else onStartPurchase();
   };
 
+  React.useEffect(
+    () => {
+      onWillPurchase();
+    },
+    // Launch the start process directly when the dialog is opened, to avoid an extra click.
+    // eslint-disable-next-line
+    []
```

Comment:
You shouldn't need to disable eslint here, if you add onWillPurchase as a dependency. It shouldn't vary in time, triggering a new call, should it?

Reply:
It will, because it's a function not wrapped in React.useCallback. I can wrap but will need to do it for everything.

In a sense, I liked the approach "Don't care what the dependencies are, just call this once and never again please" (calling it again would be a bug)